cmd/cluster-agent/custommetrics/server.go

```diff
@@ -0,0 +1,98 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package custommetrics
+
+import (
+	"os"
+	"time"
+
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/cmd/server"
+	"github.com/kubernetes-incubator/custom-metrics-apiserver/pkg/dynamicmapper"
+	"github.com/spf13/pflag"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/custommetrics"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/hpa"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var options *server.CustomMetricsAdapterServerOptions
+var stopCh chan struct{}
+
+func init() {
+	// FIXME: log to seelog
+	options = server.NewCustomMetricsAdapterServerOptions(os.Stdout, os.Stdout)
+}
+
+// AddFlags ensures the required flags exist
+func AddFlags(fs *pflag.FlagSet) {
+	options.SecureServing.AddFlags(fs)
+	options.Authentication.AddFlags(fs)
+	options.Authorization.AddFlags(fs)
+	options.Features.AddFlags(fs)
+}
+
+// ValidateArgs validates the custom metrics arguments passed
+func ValidateArgs(args []string) error {
+	return options.Validate(args)
+}
+
+// StartServer creates and start a k8s custom metrics API server
+func StartServer() error {
+	config, err := options.Config()
+	if err != nil {
+		return err
+	}
+	var clientConfig *rest.Config
+	clientConfig, err = rest.InClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(clientConfig)
+	if err != nil {
+		return log.Errorf("Unable to construct discovery client for dynamic client: %v", err)
```

Comment:
You're already logging the error for `StartServer` https://github.com/DataDog/datadog-agent/pull/1883/files#diff-5474a8acab62e3264d69daf4870ee0a9R182 so maybe you could use `fmt.Errorf` and lowercase it? (same for all `log.Errorf` below)